### PR TITLE
Update Test plan

### PR DESCRIPTION
**Pull request by ZhengJun003**
1. Some features have changed behavior.
2. Add some new feature tests.

### CODE REVIEW

File: TestPlan.md [hunk 1]
```diff
@@ -17,16 +17,26 @@
...
+### Update Effective POM file automatically
+1. Open user settings, set `true` for `maven.pomfile.autoUpdateEffectivePOM`.
+2. Change pom file and save it.
+3. Verify:
+    1. It shows `Generating effective pom ...` in status bar during generating. 
+    2. Once the operation is over, the message is cleaned from the status bar.
+4. Set `false` for `maven.pomfile.autoUpdateEffectivePOM`. 
+5. Change pom file and save it.
+6. Verify:
+    1. It not update effective pom automatically.
```
Comment (Eskibear) [inline TestPlan.md]: "It does not"

File: TestPlan.md [hunk 2]
```diff
@@ -144,9 +154,15 @@ The tree view of maven projects should update when any pom.xml is created/modifi
 3. Verify:
     1. The corresponding project item is removed from the sidebar.
 
+### Searchs for POM files
```
Comment (Eskibear) [inline TestPlan.md]: Should have better organization.
- Move this section to the end of "Basic View".   
- Change the title to "Support POM files with customized name"

File: TestPlan.md [hunk 3]
```diff
@@ -162,7 +178,8 @@ The tree view of maven projects should update when any pom.xml is created/modifi
     "maven.terminal.useJavaHome": true      // Use the Red Hat Java Language Support Setting for JAVA_HOME
     ```
 4. Trigger a maven command, in the newly created terminal, verify:
-    1. value of env `JAVA_HOME` should be the value you just set in `java.home`
+    1. On windows platform, input `$Env:MAVEN_OPTS` and press `Enter`, the value of env `JAVA_HOME` should be the value you just set in `java.home`.
```
Comment (Eskibear) [inline TestPlan.md]: `$Env:MAVEN_OPTS` only works for PowerShell. I remember it doesn't apply for CMD. 

windows => Windows

File: TestPlan.md [hunk 4]
```diff
@@ -162,7 +178,8 @@ The tree view of maven projects should update when any pom.xml is created/modifi
     "maven.terminal.useJavaHome": true      // Use the Red Hat Java Language Support Setting for JAVA_HOME
     ```
 4. Trigger a maven command, in the newly created terminal, verify:
-    1. value of env `JAVA_HOME` should be the value you just set in `java.home`
+    1. On windows platform, input `$Env:MAVEN_OPTS` and press `Enter`, the value of env `JAVA_HOME` should be the value you just set in `java.home`.
+    2. On mac/linux platform, input `echo $MAVEN_OPTS` and press `Enter`, the value of env `JAVA_HOME` should be the value you just set in `java.home`.
```
Comment (Eskibear) [inline TestPlan.md]: Mac Linux

File: TestPlan.md [hunk 5]
```diff
@@ -177,14 +194,23 @@ The tree view of maven projects should update when any pom.xml is created/modifi
         }
     ```
 2. Trigger a maven command, in the newly created terminal, verify:
-    1. value of env `MAVEN_OPTS` should be the value you just set in settings.
+    1. On windows platform, input `$Env:MAVEN_OPTS` and press `Enter`, the value of env `MAVEN_OPTS` should be the value you just set in settings.
```
Comment (Eskibear) [inline TestPlan.md]: same here.

File: TestPlan.md [hunk 6]
```diff
@@ -177,14 +194,23 @@ The tree view of maven projects should update when any pom.xml is created/modifi
...
     2. Click `Learn more`, it previews `Troubleshooting.md` in the editor.
 
+## Pom.xml file completion
+### Dependency completion
+1. In the `<dependencies>` section, type `<` and accept the suggested `dependency` maven snippet.
+2. Press `Ctrl + Space`.
+3. Verify:
+    1. The snippet is correctly inserted.
+    2. The IntelliSense is correctly worked for `<groupId>`, `<artifactId>` and `<version>`.
+
```
Comment (Eskibear) [inline TestPlan.md]: Remove the empty line here.

Comment (ZhengJun003) [reply TestPlan.md]: @Eskibear, I have updated TestPlan.md according to your comment, please have a view. Thanks.

**Pull request merged**